lib/python/rose/apps/rose_ana.py

```diff
@@ -48,6 +50,85 @@
 USRCOMPARISON_EXT = ".py"
 
 
+class KGODatabase(object):
+    """
+    KGO Database object, stores comparison information for rose_ana apps.
+
+    """
+    def __init__(self):
+        """Initialise the object"""
+        self.file_name = os.path.join(
+            os.getenv("ROSE_SUITE_DIR"), "log", "rose-ana-comparisons.db")
+        self.conn = None
+        self.create()
+
+    def get_conn(self):
+        """Return a connection to the database if it exists"""
+        if self.conn is None:
+            self.conn = sqlite3.connect(self.file_name, timeout=60.0)
+        return self.conn
+
+    def create(self):
+        """If the databaste doesn't exist, create it"""
+        self.wait_for_lock()
+        if not os.path.exists(self.file_name):
```

Comment:
You have to acquire the lock before checking if the file exists, otherwise multiple tasks could make it into the "if file doesn't exist" branch at the same time.  You'd then need an additional check just after the lock which tells the non-locking threads to bail-out... Perhaps that's better but I felt like this way around was slightly nicer?